### DISCUSSION TITLE
Add support for Res-Adapter (just released)

### DISCUSSION THREAD

**Discussion by CCpt5** [lllyasviel/stable-diffusion-webui-forge#497] [Ideas]
https://github.com/bytedance/res-adapter?tab=readme-ov-file

Would love to see this implemented into Forge/Controlnet.

_____ 

ResAdapter: Domain Consistent Resolution Adapter for Diffusion Models
Jiaxiang Cheng, Pan Xie*, Xin Xia, Jiashi Li, Jie Wu, Yuxi Ren, Huixia Li, Xuefeng Xiao, Min Zheng, Lean Fu (*Corresponding author)

AutoML, ByteDance Inc.

We propose ResAdapter, a plug-and-play resolution adapter for enabling diffusion models of arbitrary style domains to generate resolution-free images: no additional training, no additional inference and no style transfer.

**Comment by CCpt5** [comment]
The developers of "res-adapter" indicated they are working on integration w/ A1111 (Forge I assume) and Comfy. On their repo today:

https://github.com/bytedance/res-adapter/issues/1

**Comment by leomaxwell973** [comment]
so you could say... change resolution to 256x384 or 256x256 even and not have to hunt around for a good seed for an hour? :D
down for that. using those resolutions is the only way I can stand AniDiff SD1.5. ((Turbo)SDXL MMs are few, far, and fall short, so far, imo, sadly)

**Comment by CCpt5** [comment]
Update:

yesterday (March 12) they released their v1.5 model that covers up to 1024x1024. The model weight is available on HF, but still not sure how exactly to use the LoRA along w/ the "normalization" file:  
  
[2024/03/12] We release the [resadapter-sdv15](https://huggingface.co/jiaxiangc/res-adapter/tree/main/sd1.5) that supports 128~1024 resolution.  
  
https://huggingface.co/jiaxiangc/res-adapter/tree/main/sd1.5

**Comment by jiaxiangc** [comment]
@CCpt5 Hi, i am learning how to coding webui extension. I only need to load lora and normalization for sd_model.model.diffusion_model. But i do not know how to realize this in webui. Do you have any suggestions?

**Reply by CCpt5** [reply]
Edit: Nevermind, just realized YOU are the developer of Res-adapter, hahah. Thanks for your work! I'm sorry I don't have information for you, but I hope in time there is a Forge/WebUI Res-Adapter extension. If you're looking for someone to talk to w/ knowledge of Forge/A1111 in general (Re: how to load models etc), I have high regard for https://github.com/wkpark who coded an expert level model merging tool (Model MIxer), and is very activate w/ upddates/discussions on their github repos.  

Original reply where I thought you were an amateur user trying to figure Res-Adapter out:
______  

Sorry, I don't. I'm not a developer myself. The best I can point you to is a discussion on the ComfyUI node implementation which an amateur developer coded out on their own before the official group was able to share this own.  
  
See this discussion where the user "Bleeping" talks about figured this out on their own for Comfy: 

https://github.com/bytedance/res-adapter/issues/14

Thei ComfyUI node Repo:
https://github.com/blepping/ComfyUI-ApplyResAdapterUnet



if you manage to code an extension for A1111/Fogre that loads the Res-Adapter models and enables it's functionality please come back and @ me! I'd love a way to do it in the Webui.

Best of lucj!!

**Reply by CCpt5** [reply]
I commented on an old thread on @wkpark's Model Mixer repo to see if they can provide Webui model loading assistance for you. They are extremely talented/knowledgeable and friendly. I'm sure if you're seeking this assistance (and I didn't misread) they could likely help.


https://github.com/wkpark/sd-webui-model-mixer/discussions/120#discussioncomment-9035719

**Reply by jiaxiangc** [reply]
haha, thanks for your help.
